metrics/pebble_metrics.go

```diff
@@ -0,0 +1,100 @@
+package metrics
+
+import (
+	"context"
+
+	"github.com/cockroachdb/pebble"
+	"go.opentelemetry.io/otel/attribute"
+	cmetric "go.opentelemetry.io/otel/metric"
+	"go.opentelemetry.io/otel/metric/instrument"
+	"go.opentelemetry.io/otel/metric/instrument/asyncint64"
+	"go.opentelemetry.io/otel/metric/unit"
+)
+
+// pebbleMetrics asynchronously reports metrics of pebble DB
+type pebbleMetrics struct {
+	metricsProvider func() *pebble.Metrics
+	meter           cmetric.Meter
+
+	// fsyncFlushCount reports the total number of flushes
+	fsyncFlushCount asyncint64.Gauge
+	// NOTE: cache metrics report tagged values for both block and table caches
+	// cacheSize reports the number of bytes inuse by the cache
+	cacheSize asyncint64.Gauge
+	// cacheCount reports the count of objects (blocks or tables) in the cache
+	cacheCount asyncint64.Gauge
+	// cacheHits reports number of cache hits
+	cacheHits asyncint64.Gauge
+	// cacheMisses reports number of cache misses.
+	cacheMisses asyncint64.Gauge
+}
+
+func (pm *pebbleMetrics) start() error {
+	var err error
+
+	if pm.fsyncFlushCount, err = pm.meter.AsyncInt64().Gauge(
+		"ipni/dhstore/pebble/fsync_flush_count",
```

Comment:
Consistent terminology with pebble?
```suggestion
		"ipni/dhstore/pebble/flush_count",
```